Remove defunct PRENET45 from TPL Dataflow

stephentoub opened the pull request:
This PRENET45 compilation constant is legacy from when the library was building in CTPs for .NET 4, but it's not been used for a while and the library won't actually build as written with the flag set.

dnfclas commented:
Hi __@stephentoub__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.
    </span>

TTYL, DNFBOT;

weshaggard commented on src/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs:
```diff
@@ -2656,12 +2610,8 @@ private Task ProcessItemAsync(TOutput item)
-#else
-                            Task.WhenAll(_tempSendAsyncTaskList);
-#endif
+                            var allSendAsyncTasksConsolidated = Task.WhenAll(_tempSendAsyncTaskList);
```
Could you please explicitly cast this variable and a few others in in this function to Task?

terrajobst replied:
You mean replacing `var` with `Taks`, right? Casting doesn't make sense to me.

weshaggard replied:
Yes I meant to say "explicitly type" to Task. It wasn't obvious to me what the type was so I had to dig deeper.

weshaggard commented:
As I just sent you an email on if you think we will still have a .NET 4.0 version of this library in the nuget package then we should leave these and complete the code to make it work on that target. I expect that the nuget package will be completely produced by the code here once we get that fully working.

terrajobst commented:
TPL Dataflow never supported .NET 4.0, so removing the preprocessor symbol should be fine.

terrajobst commented:
LGTM

stephentoub commented:
@weshaggard, I addressed your feedback about not using var, but rather than just doing it for that one case, I went through the code and applied the guideline more rigorously.  The codebase now only uses var in cases where the type is explicitly stated on the right hand side, in particular if the right hand side is a construction, a cast, or an obvious factory.

weshaggard commented:
Thanks @stephentoub. LGTM
